From 7d5e301866267e6121f8092a21f5d0a8e5d7aa3a Mon Sep 17 00:00:00 2001
From: huntxu <mhuntxu+github@gmail.com>
Date: Thu, 22 Sep 2016 09:32:05 +0800
Subject: [PATCH 51/52] Sync #46 from devel to testing

Fix error when using fqdn as ipsec peer id (#46)

Fixes: redmine #8135

Signed-off-by: Hunt Xu <mhuntxu@gmail.com>
(cherry picked from commit ffc98bcf7c2784f2bd6e153f62d4423c55600d2b)

Signed-off-by: Hunt Xu <mhuntxu@gmail.com>
---
 neutron/services/vpn/service_drivers/ipsec.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/neutron/services/vpn/service_drivers/ipsec.py b/neutron/services/vpn/service_drivers/ipsec.py
index 9fe4806..f8ce21e 100644
--- a/neutron/services/vpn/service_drivers/ipsec.py
+++ b/neutron/services/vpn/service_drivers/ipsec.py
@@ -139,8 +139,8 @@ class IPsecVPNDriver(service_drivers.VpnDriver):
             try:
                 netaddr.IPAddress(ipsec_site_connection['peer_id'])
             except netaddr.core.AddrFormatError:
-                ipsec_site_connection['peer_id'] = (
-                    '@' + ipsec_site_connection['peer_id'])
+                ipsec_site_connection_dict['peer_id'] = (
+                    '@' + ipsec_site_connection_dict['peer_id'])
             ipsec_site_connection_dict['ikepolicy'] = dict(
                 ipsec_site_connection.ikepolicy)
             ipsec_site_connection_dict['ipsecpolicy'] = dict(
-- 
2.10.0

